src/menu-parser/parsers/ravioli.ts

```diff
@@ -12,64 +13,38 @@ const propertyMap = {
-  );
-  return data.Days.map(day => {
-    return {
-      day: moment(day.Date).format('YYYY-MM-DD'),
-      courses: day.Meals.map(meal => {
-        const properties = meal.Name.match(regExp) || [];
-        const cleanedProperties = Array.from(
-          new Set<string>(properties.map(p => p.replace(',', '')))
-        );
-        return {
-          title: `${meal.MealType}: ${meal.Name.replace(regExp, '')
-            .replace('♥', '')
-            .trim()}`,
-          properties: normalizeProperties(cleanedProperties)
-        };
-      })
-    };
-  });
-};
 
+// Today: https://menu.hus.fi/HUSAromieMenus/FI/Default/HUS/Biomedicum/Rss.aspx?Id=e60502ff-6156-4198-b0b9-a33fab86d572&DateMode=0
+// This week https://menu.hus.fi/HUSAromieMenus/FI/Default/HUS/Biomedicum/Rss.aspx?Id=e60502ff-6156-4198-b0b9-a33fab86d572&DateMode=1
+// Next week https://menu.hus.fi/HUSAromieMenus/FI/Default/HUS/Biomedicum/Rss.aspx?Id=e60502ff-6156-4198-b0b9-a33fab86d572&DateMode=2
 const parser: Parser = {
-  pattern: /lounasravintolat\.ravioli\.fi/,
+  pattern: /menu\.hus\.fi/,
   async parse(url, lang) {
-    const restaurantId = url.match(/#\/([a-zA-Z0-9\-]+)/)[1];
-    if (!restaurantId) {
-      throw new Error('Could not parse restaurant ID from URL.');
-    }
-    const restaurants = await json(
-      'http://lounasravintolat.ravioli.fi/AromiStorage/blob/main/AromiMenusJsonData'
-    );
-    const restaurantData = restaurants.Restaurants.find(
-      r => r.RestaurantId === restaurantId
-    );
-    if (!restaurantData) {
-      throw new Error('Could not find restaurant data for the id provided.');
-    }
-    const menuIds = restaurantData.JMenus.map(m => m.MenuId);
-    const allMenus = [];
-    for (const menuId of menuIds) {
-      for (const menu of await parseMenu(menuId)) {
-        const existingDayIndex = allMenus.findIndex(m => m.day === menu.day);
-        if (existingDayIndex > -1) {
-          for (const course of menu.courses) {
-            allMenus[existingDayIndex].courses.push(course);
-          }
-        } else {
-          allMenus.push(menu);
-        }
-      }
+    if (lang != 'fi') {
+      url = url.replace('/FI/', '/' + lang.toUpperCase() + '/')
     }
-    return allMenus;
+    const xml = await utils.text(url);
+    const json = await utils.parseXml(xml);
+    return json.rss.channel[0].item.map(item => {
+      var date = null
+      if (lang === 'fi') {
+        date = moment(item.title[0].split(' ')[1], 'DD.MM');
+      } if (lang === 'en') {
+        date = moment(item.title[0].split(' ')[1], 'MM/DD/YYYY')
+      } /*if (lang === 'sv') {
+        date = moment(item.title[0].split(' ')[1], 'YYYY-MM-DD')
+      }*/
+      
+      return {
+        day: date.format('YYYY-MM-DD'),
+        courses: item.description[0]
+          .split('<br>')
+          .filter(c => c.trim().length)
+          .map(title => ({ 
+            title: title, 
+            properties: []}))
```

Comment:
the properties list still seems to be always empty although properties seem to exist in the feed. to seek inspiration for parsing properties, you could take a look at some of the other parsers.

Reply:
Hi, thanks for the request, I modified the parser to properly parse the properties as well